rust/src/operations/delete.rs

```diff
@@ -0,0 +1,812 @@
+
+    let rewrite: Vec<Add> = rewrite.into_iter().map(|s| s.to_owned()).collect();
+    let parquet_scan = parquet_scan_from_actions(
+        snapshot,
+        object_store.clone(),
+        &rewrite,
+        &schema,
+        None,
+        state,
+        None,
+        None,
+    )
+    .await?;
+
+    // Apply the negation of the filter and rewrite files
+    let negated_expression = Expr::Not(Box::new(expression.clone()));
+    let physical_schema = file_schema.clone();
+    let logical_schema: DFSchema = file_schema.as_ref().clone().try_into()?;
+    let execution_props = ExecutionProps::new();
+
+    let predicate_expr = create_physical_expr(
+        &negated_expression,
+        &logical_schema,
+        &physical_schema,
+        &execution_props,
+    )?;
+    let filter: Arc<dyn ExecutionPlan> =
+        Arc::new(FilterExec::try_new(predicate_expr, parquet_scan.clone())?);
+
+    let write_start = Instant::now();
+
+    let add_actions = write_execution_plan(
+        snapshot,
+        state.clone(),
+        filter.clone(),
+        table_partition_cols.clone(),
+        object_store.clone(),
+        None,
+        None,
+        Some(writer_properties),
+    )
+    .await?;
+    metrics.rewrite_time_ms = Instant::now().duration_since(write_start).as_millis();
+
+    let read_records = parquet_scan.metrics().unwrap().output_rows().unwrap();
+    let filter_records = filter.metrics().unwrap().output_rows().unwrap();
+    metrics.num_copied_rows = Some(filter_records);
+    metrics.num_deleted_rows = Some(read_records - filter_records);
+
+    Ok((add_actions, rewrite))
+}
+
+async fn execute(
+    predicate: Option<Expr>,
+    object_store: ObjectStoreRef,
+    snapshot: &DeltaTableState,
+    state: SessionState,
+    writer_properties: WriterProperties,
+    app_metadata: Option<Map<String, Value>>,
+) -> DeltaResult<DeleteMetrics> {
+    // TODO:
+    // Metrics for records written and deleted
+    let mut metrics = DeleteMetrics::default();
+    let exec_start = Instant::now();
+
+    let (add_actions, to_delete) = match &predicate {
+        Some(expr) => {
+            let mut expr_properties = ExprProperties {
+                partition_only: true,
+            };
+
+            let current_metadata = snapshot
+                .current_metadata()
+                .ok_or(DeltaTableError::NoMetadata)?;
+
+            validate_expr(
+                expr,
+                &current_metadata.partition_columns,
+                &mut expr_properties,
+            )?;
+            excute_non_empty_expr(
+                snapshot,
+                object_store.clone(),
+                &state,
+                expr,
+                &mut metrics,
+                writer_properties,
+                expr_properties,
+            )
+            .await?
+        }
+        None => (Vec::<Add>::new(), snapshot.files().to_owned()),
+    };
+
+    let predicate = predicate.unwrap_or(Expr::Literal(ScalarValue::Boolean(Some(true))));
```

Comment:
I wanted the output of predicate to be similar to the Spark impl. For Spark, when no predicate is provided the commit entry looks like
```
{"commitInfo":{"timestamp":1682378206638,"operation":"DELETE","operationParameters":{"predicate":"[\"true\"]"},"readVersion":0,"isolationLevel":"Serializable","isBlindAppend":false,"operationMetrics":
```
My concern is if we leave if None then it would write `"predicate":null`